### DISCUSSION TITLE
Registro non funziona più

### DISCUSSION THREAD

**Discussion by sempliciniluigi89** [iisgiua/giuaschool#479] [Richieste di aiuto]
Buongiorno, dopo aver tentato nella sezione amministratore di installare i nuovi aggiornamenti successivi all'integrazione della funzione  Telegram , con scarso risultato in quanto mi usciva sempre un errore tipo di pagina non funzionante. Dopo aver lasciato perdere l'aggiornamento ,sono entrato come docente e qualsiasi cosa io clicchi mi restituisce errore 500, stessa cosa dal lato amministratore, addirittura adesso non appena mi reco sulla pagina del registro ,mi esce l'errore 500 invece della sezione di inserimento delle credenziali .

**Comment by trinko** [comment]
Non hai fatto il backup? Così ripristini la situazione precedente.
In ogni caso prova a cancellare la cache, ovvero tutto il contenuto della cartella **var/cache/prod**

**Comment by sempliciniluigi89** [comment]
No, purtroppo non ho pensato proprio a fare il backup ,in ogni caso ho provato a cancellare tutto il contenuto della cartella var/cache/prod , ma la situazione resta invariata

**Reply by trinko** [reply]
Prova a caricare il file di log, perché forse c'è qualche informazione sull'errore.

**Reply by sempliciniluigi89** [reply]
Il problema è che non riesco proprio ad accedere al registro ,non appena clicco sul link dello stesso registro ,mi esce l'errore 500,ragione per la quale non riesco ad accedere e caricare il file log

**Reply by trinko** [reply]
Il file di log lo puoi scaricare tramite il programma FTP dalla cartella **var/log/prod/**. Prendi il log di oggi e, possibilmente, taglia la parte non rilevante.

**Comment by trinko** [comment]
Sembra che manchino dei file: probabilmente l'aggiornamento si è interrotto senza completare la modifica di alcuni file di sistema.
Allego di seguito un file che viene indicato negli errori. Prova quindi:

1) estrai il file seguente e caricalo nello stesso percorso indicato dalle cartelle compresse; prova a vedere se è cambiato qualcosa.

[vendor.zip](https://github.com/iisgiua/giuaschool/files/11056919/vendor.zip)

2) se non è migliorata la situazione, è necessario assicurarsi di caricare tutti i file, tramite FileZilla o altro programma, dell'ultimo aggiornamento. 
[L'aggiornamento è in questo file](https://github.com/iisgiua/giuaschool/releases/download/update-v1.5.2/giuaschool-update-v1.5.2.zip)

3) cancella a mano la cache (cartella **var/cache/prod**) e vedi se funziona.

**Reply by sempliciniluigi89** [reply]
Buonasera,seguendo il punro 2. sono riuscito a riportare il tutto come prima,fortunatamente.Solo che andando in aggiornamento mi continua ad apparire il messaggio di aggiornamento:L'aggiornamento alle modifiche successive alla versione 1.5.2 è stato scaricato correttamente.
Si ricorda che tali modifiche non sono ancora validate né definitive, per cui se ne sconsiglia l’installazione se non in ambiente di test.

Prima di proseguire con l'installazione, assicurati di aver eseguito il backup del sistema e di aver attivato la modalità manutenzione.


solo che quando clicco su contia ci mette un eternità e poi mi restituisce il messaggio che la pagina non funziona

**Comment by sempliciniluigi89** [comment]
Adesso però,mi si presenta un altro problema legato sicuramente all'aggiornamento non andato a buon fine.Nono appena l'amministratore  accede nel registro  e si reca sul profilo per poi cliccare su notifiche ,appare l'errore 500 che allego 
[2023-03-24_15-59.log](https://github.com/iisgiua/giuaschool/files/11063597/2023-03-24_15-59.log)

**Comment by trinko** [comment]
Il problema è dovuto al fatto che la procedura di installazione non ha fatto le necessarie modifiche al database.
L'errore che non ti fa completare l'installazione credo sia dovuto al tempo di scaricamento dell'aggiornamento che va oltre il limite previsto dal tuo hosting.

Puoi provare a saltare i passaggi dell'installazione che creano problemi nel modo seguente.
1. crea un nuovo file (anche vuoto, non importa), che abbia come nome **giuaschool-update-v1.5.2.ok**
2. comprimini con ZIP il file precedente e dagli come nome **giuaschool-update-v1.5.2.zip**
3. carica con FileZilla entrambi i due file ( **giuaschool-update-v1.5.2.ok** e **giuaschool-update-v1.5.2.zip**) all'interno della cartella **src/Install**

A questo punto riprova a far partire l'aggiornamento.

**Reply by sempliciniluigi89** [reply]
Ho provato ma l'errore è sempre il seguente:La pagina non funzionascuoladigitalelatiano.x10.mx al momento non è in grado di gestire la richiesta.
HTTP ERROR 500

**Comment by trinko** [comment]
In tal caso credo che l'errore dipenda da una configurazione particolare del tuo servizio di hosting.

Se vuoi, puoi comunque eseguire a mano le modifiche al database:

1.  vai sulla gestione database, tramite PhpMyadmi
2. apri la scheda per i comandi SQL
3. inserisci i seguenti comandi (una riga alla volta) e premi il pulsante di esecuzione:
```
UPDATE gs_utente SET notifica='a:2:{s:4:\"tipo\";s:5:\"email\";s:9:\"abilitato\";a:2:{i:0;s:6:\"avviso\";i:1;s:9:\"circolare\";}}' WHERE notifica IS NULL OR notifica='a:0:{}';
UPDATE gs_utente SET rappresentante='' WHERE rappresentante IS NULL OR rappresentante='a:0:{}';
UPDATE gs_configurazione SET valore='17:00',descrizione='Ora di inizio notifica giornaliera delle nuove circolari\<br>[ora in formato HH:MM]' WHERE parametro='notifica_circolari';
```

4. rifai la stessa con questi altri comandi; ignora eventuali errori del tipo _chiave duplicata_,  dovuti al fatto che la modifica era già stata effettuata.
```
INSERT INTO gs_configurazione (creato, modificato, categoria, parametro, descrizione, valore, gestito) VALUES (NOW(), NOW(), 'SISTEMA', 'telegram_bot', 'Indica il nome del bot Telegram usato per l\'invio delle notifiche\<br>[testo]', '', 1);
INSERT INTO gs_configurazione (creato, modificato, categoria, parametro, descrizione, valore, gestito) VALUES (NOW(), NOW(), 'SISTEMA', 'telegram_token', 'Indica il token del bot Telegram usato per l\'invio delle notifiche\<br>[testo]', '', 1);
INSERT INTO gs_configurazione (creato, modificato, categoria, parametro, descrizione, valore, gestito) VALUES (NOW(), NOW(), 'SISTEMA', 'telegram_secret', 'Indica il codice segreto usato dal bot Telegram per l\'esecuzione della procedura di registrazione\<br>[testo]', '', 1);
INSERT INTO gs_configurazione (creato, modificato, categoria, parametro, descrizione, valore, gestito) VALUES (NOW(), NOW(), 'SISTEMA', 'comando_token', 'Indica il codice segreto usato per l\'esecuzione di un comando tramite URL\<br>[testo]', '', 0);
INSERT INTO gs_configurazione (creato, modificato, categoria, parametro, descrizione, valore, gestito) VALUES (NOW(), NOW(), 'SISTEMA', 'versione_build', 'Numero di build dell\'applicazione<br>[testo]', '0', 1);
```

5. infine esegui questi ultimi comandi; ignora eventuali messaggi di errore di duplicazione o chiave già esistente.

```
CREATE TABLE IF NOT EXISTS gs_presenza (id INT AUTO_INCREMENT NOT NULL, alunno_id INT NOT NULL, creato DATETIME NOT NULL, modificato DATETIME NOT NULL, data DATE NOT NULL, ora_inizio TIME DEFAULT NULL, ora_fine TIME DEFAULT NULL, tipo VARCHAR(1) NOT NULL, descrizione VARCHAR(255) NOT NULL, INDEX IDX_32C5F7A67ABC9740 (alunno_id), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8 COLLATE `utf8_unicode_ci` ENGINE = InnoDB;
ALTER TABLE gs_presenza ADD CONSTRAINT FK_32C5F7A67ABC9740 FOREIGN KEY (alunno_id) REFERENCES gs_utente (id);
ALTER TABLE gs_messenger_messages CHANGE id id BIGINT AUTO_INCREMENT NOT NULL, ADD PRIMARY KEY (id);
```

**Comment by trinko** [comment]
Dimenticavo: alla fine cancella la cache.

**Reply by sempliciniluigi89** [reply]
Non ho risolto niente ,non appena vado in aggiornamento ,l'avviso è sempre il seguente:L'aggiornamento alle modifiche successive alla versione 1.5.2 è stato scaricato correttamente.
Si ricorda che tali modifiche non sono ancora validate né definitive, per cui se ne sconsiglia l’installazione se non in ambiente di test.

Prima di proseguire con l'installazione, assicurati di aver eseguito il backup del sistema e di aver attivato la modalità manutenzione.

**Comment by sempliciniluigi89** [comment]
La cosa strana è che fino a pochi giorni, mesi ho eseguito tutti gli aggiornamenti senza nessun problema

**Comment by trinko** [comment]
Ma il problema della pagina delle notifiche è risolto?

Per riprovare l'aggiornamento, devi cancellare i file che avevi caricato in precedenza: 
- **src/Install/giuaschool-update-v1.5.2.zip**
- **src/Install/giuaschool-update-v1.5.2.ok**

Ma se ho capito il problema, si ripresenterà un errore in fase di scaricamento dell'aggiornamento, dovuto al fatto che ci mette troppo tempo a scaricarlo (e alle impostazioni del tempo massimo previsto dal tuo servizio di hosting). In precedenza non dava problemi perché il pacchetto di aggiornamento era molto più piccolo.

**Reply by sempliciniluigi89** [reply]
Purtroppo non si è risolto niente, per quanto riguarda il problema delle notifiche dal lato amministratore , è rimasto invariato

**Comment by trinko** [comment]
Per risolvere il problema della pagina notifiche, prova ad eseguire questo comando SQL:

```
UPDATE gs_utente SET notifica='a:2:{s:4:\"tipo\";s:5:\"email\";s:9:\"abilitato\";a:2:{i:0;s:6:\"avviso\";i:1;s:9:\"circolare\";}}';
```

**Reply by sempliciniluigi89** [reply]
Purtroppo non si è risolto
